### ISSUE TITLE
Exception when using Textures already in Assets Folder

### ISSUE THREAD

**Issue by MeFisto94** [xbuf/blender_io_xbuf#10]
The Issue is here:
https://github.com/xbuf/blender_io_xbuf/blob/master/xbuf_export.py#L767

shutils.copy raises an Exception when the files are identical. We DO check the paths for equality but don't properly resolve them (We'd need the absolute path for that).

We could now either catch that Exception or check for an absolute path. The pythonic way would definately be to catch the exception. `SameFileError` of shutils.

We should also add something like `if !file.exists(target)`, but then we should check if the hash differs, and if the hash differs, delete/overwrite the target file OR Fail?

The comment also says: `# TODO use md5 (hashlib.md5().update(...)) to name or to check change ??`

**Comment by MeFisto94** [comment]
If desired, I could make a pull request with the following (pseudocode as of now):

```
if file.exists(target):
  if hashlib.md5.update(target) == hashlib.md5.update(source):
     return
  else:
     raise Error("Already have a texture of name %s which differs from the one for the xbuf file. Resolve that Name Clash!" % (target_name))
```

That way we don't overwrite already existing textures and when there is a Name Clash (multiple Stone.jpg Textures), the User is warned as overwriting would lead to an undefined behavior.

**Comment by davidB** [comment]
PR are welcomes.
But I'm not sur that computing md5 is quicker than "always" copying the
file (highly optimized by FS).
Overwite force the user to have what he seen (preview,...)

**Comment by MeFisto94** [comment]
Well it is not really much quicker (since it has to read both files) and if the files are not equal you even have to copy them anyway.
It only saves write cycles which might be interesting for SSDs, but it feels like optimizing unnecessary stuff.
It is faster though when you see the files as unequal if the filesize is different.

Regarding the overwrite: When you have multiple models which accidentally have the same texture name but different textures then always the lastly exported one dominates.

**Comment by MeFisto94** [comment]
So I tried to apply a fix but that leads to another fault due to the new TBN Generation, which is only there in master.

`dst.attrib = xbuf.datas_pb2.VertexArray.tbn_to_model_quat` (Line 555) has no tbn_to_model_quat defined, so I commented out the whole export tbns method currently.

I tried a `except` there but then the serialization fails, possibly due to line 554.
I'll see if I can open the xbuf file and open a Pull Request for this Issue soon

Edit: I have another issue (at least with the sdk) which used to work with the protobuf 2 xbuf version.

```
com.jme3.asset.AssetNotFoundException: WrapRepeat_S WrapRepeat_T "Textures/BarkMossy-tiled-c.png" (Flipped)
    at com.jme3.asset.DesktopAssetManager.loadAsset(DesktopAssetManager.java:370)
    at com.jme3.asset.DesktopAssetManager.loadTexture(DesktopAssetManager.java:390)
    at com.jme3.gde.core.properties.preview.TexturePreview$1.call(TexturePreview.java:97)
    at com.jme3.gde.core.properties.preview.TexturePreview$1.call(TexturePreview.java:92)
```

It might have confused the Wrapping Mode as Asset? I have to check deeper since manually changing the texture in the material didn't yet lead to success
